source/lex.tex

```diff
@@ -288,6 +288,25 @@
 The \grammarterm{universal-character-name} construct provides a way to name
 other characters.
 
+\begin{bnf}
+\nontermdef{n-char}: one of\br
+    A B C D E F G H I J K L M N O P Q R S T U V W X Y Z\br
+    0 1 2 3 4 5 6 7 8 9\br
+    \unicode{002d}{hyphen-minus}\br
+    \unicode{0020}{space}\br
+\end{bnf}
+
+\begin{bnf}
+\nontermdef{n-char-sequence}:\br
+    n-char\br
+    n-char-sequence n-char
+\end{bnf}
+
+\begin{bnf}
+\nontermdef{named-universal-character}:\br
+    \terminal{\textbackslash N} {n-char-sequence}\br
```

Comment:
The braces around _n-char_sequence_ don't make it through LaTeX (braces are special for LaTeX). Please use `\terminal{\{}`.